anyblok/bloks/anyblok_core/tests/test_system_sequence.py

```diff
@@ -29,66 +41,73 @@ def test_nextval_without_prefix_without_suffix_two_time(
         registry = rollback_registry
         Sequence = registry.System.Sequence
         seq = Sequence.insert(code='test.sequence', formater='prefix_{seq}')
-        number = seq.number
-        assert seq.nextval() == 'prefix_%d' % (number + 1)
-        assert seq.nextval() == 'prefix_%d' % (number + 2)
-        assert seq.nextval() == 'prefix_%d' % (number + 3)
+        current = seq.current
+        assert current is None
+        assert seq.nextval() == 'prefix_%d' % 1
+        assert seq.nextval() == 'prefix_%d' % 2
+        assert seq.nextval() == 'prefix_%d' % 3
 
     def test_nextval_with_prefix_with_suffix(self, rollback_registry):
         registry = rollback_registry
         Sequence = registry.System.Sequence
         seq = Sequence.insert(code='test.sequence',
                               formater='prefix_{seq}_suffix')
-        number = seq.number
-        assert seq.nextval() == 'prefix_%d_suffix' % (number + 1)
-        assert seq.nextval() == 'prefix_%d_suffix' % (number + 2)
-        assert seq.nextval() == 'prefix_%d_suffix' % (number + 3)
+        current = seq.current
+        assert current is None
+        assert seq.nextval() == 'prefix_%d_suffix' % 1
+        assert seq.nextval() == 'prefix_%d_suffix' % 2
+        assert seq.nextval() == 'prefix_%d_suffix' % 3
 
     def test_nextval_by_attribute(self, rollback_registry):
         registry = rollback_registry
         Sequence = registry.System.Sequence
         seq = Sequence.insert(code='test.sequence')
-        number = seq.number
-        assert Sequence.nextvalBy(code=seq.code) == str(number + 1)
-        assert Sequence.nextvalBy(code=seq.code) == str(number + 2)
-        assert Sequence.nextvalBy(code=seq.code) == str(number + 3)
+        current = seq.current
+        assert current is None
+        assert Sequence.nextvalBy(code=seq.code) == '1'
+        assert Sequence.nextvalBy(code=seq.code) == '2'
+        assert Sequence.nextvalBy(code=seq.code) == '3'
 
     def test_sequence_with_gap_rollback(self, rollback_registry):
         registry = rollback_registry
         Sequence = registry.System.Sequence
         seq = Sequence.insert(code='test.sequence', no_gap=False)
-        number = seq.number
-        assert Sequence.nextvalBy(code=seq.code) == str(number + 1)
+        current = seq.current
+        assert current is None
+        assert Sequence.nextvalBy(code=seq.code) == '1'
         registry.commit()
-        assert Sequence.nextvalBy(code=seq.code) == str(number + 2)
+        assert Sequence.nextvalBy(code=seq.code) == '2'
         registry.rollback()
-        assert Sequence.nextvalBy(code=seq.code) == str(number + 3)
+        assert Sequence.nextvalBy(code=seq.code) == '3'
 
     def test_sequence_with_no_gap_rollback(self, rollback_registry):
         registry = rollback_registry
         Sequence = registry.System.Sequence
         seq = Sequence.insert(code='test.sequence.nogap', no_gap=True)
```

Comment:
I haven't seen a test with ``no_gap=True`` and ``start=10`` could be nice to cover this use case !